pkg/morph/client/notary.go

```diff
@@ -164,6 +165,32 @@ func (c *Client) DepositNotary(amount fixedn.Fixed8, delta uint32) (res util.Uin
 		till = currentTill
 	}
 
+	return c.depositNotary(amount, till)
+}
+
+// DepositEndlessNotary calls notary deposit method. Unlike `DepositNotary`,
+// this method sets notary deposit until parameter to a maximum possible value.
+// This allows to avoid ValidAfterDeposit failures.
+//
+// This function must be invoked with notary enabled otherwise it throws panic.
+func (c *Client) DepositEndlessNotary(amount fixedn.Fixed8) (res util.Uint256, err error) {
+	c.switchLock.RLock()
+	defer c.switchLock.RUnlock()
+
+	if c.inactive {
+		return util.Uint256{}, ErrConnectionLost
+	}
+
+	if c.notary == nil {
+		panic(notaryNotEnabledPanicMsg)
+	}
+
+	// MaxUint64 fails transfer transaction
+	// failed native call: at instruction 36 (SYSCALL): bigint does not fit into uint32
```

Comment:
I think it would be more descriptive to mention that `till` refer to some block height and it is `uint32` in neo-go.